src/main/java/by/exadel/internship/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package by.exadel.internship.service.impl;
+
+import by.exadel.internship.service.FileService;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import java.util.UUID;
+
+@Service
+public class FileServiceImpl implements FileService {
+
+    private static final String DOWNLOAD_URL =
+            "https://firebasestorage.googleapis.com/v0/b/gs://internship-project-e202a.appspot.com/o/%s?alt=media";
+    private static final String BUCKET_NAME = "internship-project-e202a.appspot.com";
+
+    @Override
+    public ResponseEntity<String> upload(MultipartFile multipartFile) {
```

Comment:
I don't think ResponseEntity is good return type for service. You are mixing architecture levels and actually configure response here

Reply:
Done